more_autograding_examples/jupyter_notebook_autograding/config/test_input/grader.py

```diff
@@ -0,0 +1,93 @@
+import os
+import sys
+import nbformat
+from nbconvert.preprocessors import ClearOutputPreprocessor, ExecutePreprocessor, CellExecutionError
+from nbconvert import HTMLExporter
+import matplotlib.pyplot as plt
+import matplotlib.image as img
+import base64
+
+# Execution API: https://nbconvert.readthedocs.io/en/latest/execute_api.html
+def execute_notebook(notebook_path, timeout=600):
+
+    notebook_filename = os.path.split(notebook_path)[-1]
+    notebook_filename_out = ''.join(notebook_filename.split('.')[0]) + '_executed.ipynb'
```

Comment:
This is pretty ugly.  You could use `pathlib`'s [`.stem` ](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem) instead.